src/__tests__/labs/ViewLabs.test.tsx

```diff
@@ -125,30 +120,36 @@ describe('View Labs', () => {
         screen.getByRole('combobox'),
         `{selectall}{backspace}${expectedStatus}{arrowdown}{enter}`,
       )
-
-      expect(LabRepository.search).toHaveBeenCalledTimes(2)
-      expect(LabRepository.search).toHaveBeenCalledWith(
-        expect.objectContaining({ status: expectedStatus }),
-      )
+      expect(screen.getByRole('combobox')).toHaveValue('labs.status.requested')
     })
   })
 
   describe('search functionality', () => {
     it('should search for labs after the search text has not changed for 500 milliseconds', async () => {
+      const expectedLab2 = {
+        code: 'L-5678',
+        id: '5678',
+        type: 'another type',
+        patient: 'patientIdB',
+        status: 'requested',
+        requestedOn: '2020-03-30T04:43:20.102Z',
+      } as Lab
+      const expectedSearchText = 'another'
+      const { expectedLab } = setup([Permissions.ViewLabs, Permissions.RequestLab])
+
+      jest.spyOn(LabRepository, 'findAll').mockResolvedValue([expectedLab2])
+
+      expect(await screen.findByRole('cell', { name: expectedLab.code })).toBeInTheDocument()
       jest.useFakeTimers()
-      setup([Permissions.ViewLabs])
 
-      const expectedSearchText = 'search text'
       userEvent.type(screen.getByRole('textbox', { name: /labs.search/i }), expectedSearchText)
 
       act(() => {
         jest.advanceTimersByTime(500)
       })
```

Comment:
Is there a way to do this with `waitFor()` the expect or something?

Reply:
Looks like it will pass if I replace `act` with `await waitFor`, is that what you're talking about?

Reply:
I was thinking more like
```suggestion
      await wairFor(() => {
        expect(await finBy*(whatever))
      })
```

Reply:
An await in an await? :sweat_smile: 

Either:
```javascript
await waitFor(() => {
  expect(screen.getBy*())
}
```
Or
```javascript
expect(await screen.findBy*())
```

Reply:
what never seen async/await hell lmao